applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_implicit_dynamic_solver.py

```diff
@@ -73,23 +73,30 @@ def _create_solution_scheme(self):
         elif(scheme_type == "bossak"):
             damp_factor_m = self.dynamic_settings["damp_factor_m"].GetDouble()
             mechanical_scheme = KratosMultiphysics.ResidualBasedBossakDisplacementScheme(damp_factor_m)
-        elif("bdf" in scheme_type or scheme_type == "backward_euler"):
-            if (scheme_type == "backward_euler"):
-                order = 1
-            else:
-                order = int(scheme_type.replace("bdf",""))
-            
-            # Warning
-            if (order > 2):
-                KratosMultiphysics.Logger.PrintWarning("WARNING:: BDF Order: ", str(order) + " constant time step must be considered")
+        elif(scheme_type.startswith("bdf") or scheme_type == "backward_euler"):
+            order = self._bdf_integration_order()
             mechanical_scheme = KratosMultiphysics.ResidualBasedBDFDisplacementScheme(order)
         elif(scheme_type == "relaxation"):
             damp_factor_f =-0.3
             dynamic_factor_m = 10.0
             mechanical_scheme = StructuralMechanicsApplication.ResidualBasedRelaxationScheme(
                                                                        damp_factor_f, dynamic_factor_m)
         else:
-            err_msg =  "The requested scheme type \"" + scheme_type + "\" is not available!\n"
-            err_msg += "Available options are: \"newmark\", \"bossak\", \"backward_euler\", \"bdf2\", \"relaxation\""
+            err_msg = "The requested scheme type \"" + scheme_type + "\" is not available!\n"
+            err_msg += "Available options are: \"newmark\", \"bossak\", \"backward_euler\", \"bdf1\", \"bdf2\", \"bdf3\", \"bdf4\", \"bdf5\", \"relaxation\""
             raise Exception(err_msg)
         return mechanical_scheme
+
+    def _bdf_integration_order(self):
+        scheme_type = self.dynamic_settings["scheme_type"].GetString()
+        if (scheme_type == "backward_euler"):
+            order = 1
+        else:
+            # BDF schemes can be from 1 to 5 order, so in order to detect the integration order from the scheme_type we remove the "bdf" string, that is, if the user tells bdf3 only 3 will remain when we remove bdf which corresponds to the method of choice
+            order = int(scheme_type.replace("bdf", ""))
+
+        # Warning
+        if (order > 2):
```

Comment:
do you still have the requirement of using a constant time step??

Reply:
Just for order 3 and above (1 and 2 is OK)

Reply:
Thats why there is order > 2